docs/source/user-guide.rst

```diff
@@ -194,12 +196,16 @@ Writing NRRD files
 ------------------
 Writing to NRRD files can be done with the single function :meth:`write`. The :obj:`filename` parameter to the function specifies the absolute or relative filename to write the NRRD file. If the :obj:`filename` extension is .nhdr, then the :obj:`detached_header` parameter is set to true automatically. If the :obj:`detached_header` parameter is set to :obj:`True` and the :obj:`filename` ends in .nrrd, then the header file will have the same path and base name as the :obj:`filename` but with an extension of .nhdr. In all other cases, the header and data are saved in the same file.
 
-The :obj:`data` parameter is a :class:`numpy.ndarray` of data to be saved. :obj:`header` is an optional parameter of type :class:`dict` containing the field/values to be saved to the NRRD file. 
+The :obj:`data` parameter is a :class:`numpy.ndarray` of data to be saved. :obj:`header` is an optional parameter of type :class:`dict` containing the field/values to be saved to the NRRD file. :obj:`index_order` specifies the index order of the array :obj:`data`, this should be consistent with the argument of the same name in :meth:`read`.
 
 .. note::
 
     The following fields are automatically generated based on the :obj:`data` parameter ignoring these values in the :obj:`header`: 'type', 'endian', 'dimension', 'sizes'.
 
 .. note::
 
-    The default encoding field used if not specified in :obj:`header` is 'gzip'.
\ No newline at end of file
+    The default encoding field used if not specified in :obj:`header` is 'gzip'.
+
+.. note::
+
```

Comment:
I think there's a risk of just generating more confusion by doing this. For instance, you can't really go into this ordering business without actually discussing memory layout. Phrases like "fastest-varying to slowest-varying." means nothing without the association to memory.

In my opinion, we should probably stick to a simple description, i.e. (`x,y,z` vs `z,y,x`). Feels a bit dishonest since the NRRD specification doesn't assume any names for the axes but it should be clear enough.

Reply:
I can understand that. It was difficult trying to write this without going into detail of the memory layout. However, I like the idea of having an `Index Ordering` section because it ties together the argument for the read/write functions and it has a number of other pieces of info:
1. C-order vs. Fortran-order is just a transpose in the data
2. Index order **only** affects the data, not the header fields. Header fields are always in Fortran-order.
3. Explains that Python typically uses C-order. Fortran-order is the default but will change in the future.

Do you have a better way to describe the difference between C-order and Fortran-order? My hesitation to **just** use (x, y, z) as the example for describing then it is not obvious that the order for an RGB image is (ch, x, y, z).

We could have a table of example coordinate conventions for Fortran-order and C-order, like scikit-image has: http://scikit-image.org/docs/dev/user_guide/numpy_images.html#coordinate-conventions

What are your thoughts?

**Edit:** Whatever wording we use for describing the index order, I think that should propagate to the read & write docstrings.

Reply:
I added the index ordering section and made an attempt to provide a brief description of the data layout.